rootfs/etc/nginx/lua/balancer/sticky.lua

```diff
@@ -31,20 +33,62 @@ local function encrypted_endpoint_string(self, endpoint_string)
   return encrypted
 end
 
+local function parse_cookie_expires(expires)
+  local time = tonumber(string.sub(expires, 0, string.len(expires) - 1))
+  if time == nil then
+    return nil, string.format("the time of expires (%s) is wrong", expires)
+  end
+
+  local unit = string.sub(expires, -1)
+  if unit == "y" then
+    return time * 60 * 60 * 24 * 365
+  elseif unit == "M" then
+    return time * 60 * 60 * 24 * 30
+  elseif unit == "w" then
+    return time * 60 * 60 * 24 * 7
+  elseif unit == "d" then
+    return time * 60 * 60 * 24
+  elseif unit == "h" then
+    return time * 60 * 60
+  elseif unit == "m" then
+    return time * 60
+  elseif unit == "s" then
+    return time
+  else
+    return nil, string.format("the unit of expires (%s) is wrong", expires)
+  end
+end
```

Comment:
Can you parse this in controller side? I bet `time` package has something built-in you can use.

Reply:
Yes, the `time` package has a function `ParseDuration` that can parses a duration string, but its valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h". For our case, we  plan to handle "y", "M", "w", "d",  "h", "m", "s", so I think it is not enough for us.

Reply:
@skeeey what if we ask user to set unix timestamp as the value of expires annotation? That way we would not have to deal with any of these

Reply:
@ElvinEfendi Yes, if user use unix timestamp as the value of expires, we will not deal with any of these, but I think this is not friendly for a user. e.g. If one user want to make his cookie expire in one day, the user have to  calculate the unix timestamp that he wanted, rather than, just give a simple value 1d.

And if you think we should not parse the time in lua part (this may make lua code a bit complicated) I can put this in controller side (add an enhanced `ParseDuration` function to `k8s.io/ingress-nginx/internal/ingress/annotations/parser`)

What is your opinion?

Reply:
@ElvinEfendi Yes, you are right, current code cannot parse the year or moth accurately, if we want to handle them, it will be complicated.  so I use second as the `expires` unit like the `max-age`,  what is your opinion?

Reply:
> so I use second as the expires unit like the max-age, what is your opinion?

👍